### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2018-03-01

### PR DESCRIPTION
**Pull request by yashmuel**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/yashmuel/azure-rest-api-specs/dev-monitor-Microsoft.Insights-2018-03-01-yashmuel-fix-swagger-validations...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
Comment (asafst) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:324]: even though this is only a description, this is not only for bad request

**Review by asafst — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: It seems not a good practice to refer to another version (but as it is already done, there might not possible to fix within this version).

Also the response schema is not correct as in LintDiff. It should have an error property, which then contains code and message.
Though it is likely not possible to fix within same version, you might need to suppress.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft right.
I've requested a suppression for the LintDiff and also mentioned it in the swagger.
What about the sdk check which fails, what should i do with this? suppression as well?

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: You can ignore the CI that without "required" label. They does not block the merge.

As dotnet build, it appears from its log above, fails on some class as `azure-sdk-for-net/sdk/monitor/Microsoft.Azure.Management.Monitor/tests/BasicTests/DataCollectionRulesTests.cs`, so it is likely not caused by your PR.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft  cool, so is there anything that blocking me to merge this PR?

Another thing, I've another PR (https://github.com/Azure/azure-rest-api-specs/pull/12234) with some error about Swagger ApiDocPreview failed, should i fixed this or it's like the sdk?

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: I tried to re-run ApiDocPreview CI in the other PR.

Both PR still have LintDiff, which blocks. I think perhaps the suppress is not done correctly. The correct one is perhaps
```
  - suppress: R4007
```

**Comment by weidongxu-microsoft** [comment]
PrettierCheck fails.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: I think you're right, fixed this and now the LintDiff passed

**Comment by yashmuel** [comment]
> PrettierCheck fails.

fixed this.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft  btw, i've few more PRs which waiting a long time, can you please take a look also?
https://github.com/Azure/azure-rest-api-specs/pull/12232
https://github.com/Azure/azure-rest-api-specs/pull/12233
https://github.com/Azure/azure-rest-api-specs/pull/9790

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: For https://github.com/Azure/azure-rest-api-specs/pull/9790 you might again ping the owner.

For the rest on me I will merge them. However after first PR merge there is conflict on the other PRs. You need to resolve the conflict as I do not have permission to commit to your forked repo.

Please resolve 1 PR, ping me, I will merge it. Then proceed on next. They all changes the same readme.md and hence got conflict against each other.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: cool, saw you merged one. resolved conflict here: https://github.com/Azure/azure-rest-api-specs/pull/12234
can you please merge?

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: done 12234

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: thanks, resolved https://github.com/Azure/azure-rest-api-specs/pull/12233
merge?

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: 12233 still has the conflict. You probably need to refresh master every time before merge it to branch.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: or just click commit after resolving :)
make sure i resolved it correctly and you can merge it.

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: 12233 merged

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: great, thanks!
so this is the only one left (and also the old one which you're not own)
i have to wait a bit with this one, cause maybe we need to fix a small thing, so i guess it will be tomorrow or next week.
i'll ping you here :)

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: This PR 12231 could also be resolved and merged.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: yea, i'm talking about this one which i have to be a small change in it before merging

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft resolved conflict in current PR and we can merge it as well :)
btw, what can i do with the other PR that i've that the reviewer doesn't answer after 2 pings?
https://github.com/Azure/azure-rest-api-specs/pull/9790

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: "Swagger ModelValidation" failed on this. I could not get the cause. I will check around.

I've merge the other one.

**Comment by weidongxu-microsoft** [comment]
@akning-ms Hi Arthur, could you help merge this PR? Jeffrey approved the breaking changes.

I've checked with Ray Chen, that fail on "ModelValication" is false alert (the swagger has a local `Resource` type, and also loads the `Resource` from common-types, hence the confusion).

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft  Thanks a lot for all that work!
Can you please take a look at this one? https://github.com/Azure/azure-rest-api-specs/pull/12314
I can't figure out what's the issue with the LintDiff

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: Not sure. My guess is that if you have multiple json, and they all defines `ApiVersionParameter`, you need to make it identical includes its `description`.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft thanks, i'm not sure that was a real problem, but solve this and now there is another issue that i'm not sure how to deal with, can you please take another look? 
https://github.com/Azure/azure-rest-api-specs/pull/12314

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: [microsoft.insight.txt](https://github.com/Azure/azure-rest-api-specs/files/5802257/microsoft.insight.txt)
attached our operations API which contains this API, not sure why this validation fails

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: I think it asks for `operations` endpoint like this https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/operation.json#L37?

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft We haven't operations_API.json file in any of our working APIs, for example this one is the same API with different version and without operations_API.json file and we've completed a PR on this one last week:
https://github.com/Azure/azure-rest-api-specs/tree/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01
do you have any idea who can i ask such questions? i'm blocked on this issue :/

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: Here is the requirement https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r3023

Some validation only apply to new api-version.

You can check with "akning-ms" on this.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: what is akning-ms and how can i contact them?

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: This is his github ID. Though I do not think it will be exempted. And monitor had the json https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/operations_API.json

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft 
cool, i'll try him.

I've 2 additional PRs that i want to merge, but experiencing the same model validations we had in the current PR,
can you do the same process you did for this one to merge them?
https://github.com/Azure/azure-rest-api-specs/pull/12371
https://github.com/Azure/azure-rest-api-specs/pull/12372

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: For these you can also contact Arthur.

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: sure, will do.
but first you don't have to verify that the failing model validation are false alerts?

Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json:326]: @weidongxu-microsoft talked with Ray and suppressed the modelValidation, can you please help me merge this one?
https://github.com/Azure/azure-rest-api-specs/pull/12371

**Pull request merged**
